### PR TITLE
Fix: XP earned by Tag no longer shows unapproved quests; Closes #1208

### LINKED ISSUE #1208: XP earned by Tag shows unapproved quests on Profile page
However it shows as 0 XP, and doesn't appear on the chart either, so data is accurate, just a bit confusing that it shows up at all.

image

### PR DESCRIPTION
**Pull request by 9921622**
Please ensure you are familiar with our Pull Request Description expectations described here: https://www.pullrequest.com/blog/writing-a-great-pull-request-description/
### What?
Fixed an issue where submitted but not approved quest show in "XP Earned by Tag" column in profile detail page
This is an issue because "XP Earned by Tag" indicates anthing in that table is xp that was earned. Submitted but not approved quests do not give xp. Therefore this is an error.

### Why?
See #1208

### How?
Tweaking one the query filters in get_tags_from_user fixed this. Specifically adding an is_approved=True.
image

### Testing?
Created a test case where 3 unique quests linked to a tag, with 2 unique submissions were created.
Only 1 of the 3 quests was approved.

The test will only pass if that one quest is in get_tags_from_user

### Screenshots (if front end is affected)
unapproved quest with screenshot tag
image
image

### Anything Else?
None

### Review request
@tylerecouture

### CODE REVIEW

File: src/tags/tests/test_models.py [hunk 1]
```diff
@@ -179,6 +179,30 @@ def test_multiple_tags_per_quest_badges(self):
...
+        q_approved.tags.add('approved')
+        q_submitted.tags.add('submitted')
+
+        # check if only q_approved is in the query
+        tags = [tag_tuple[0].name for tag_tuple in get_user_tags_and_xp(self.user)]
+
+        self.assertFalse('unrelated' in tags)
+        self.assertTrue('approved' in tags)
+        self.assertFalse('submitted' in tags)
+
```
[the remark this answers is not in the document]
Comment (9921622) [reply src/tags/tests/test_models.py:205]: @coderabbitai sure

[the remark this answers is not in the document]
Comment (9921622) [reply src/tags/tests/test_models.py:205]: i dont think this is necessary, this was already covered in other test cases. im resolving

**Comment by 9921622** [comment]
@tylerecouture review

**Review by tylerecouture — CHANGES_REQUESTED** [review]
Your "WHY" belongs in the WHAT and your WHY is missing link to issue.  Please fix.

**Comment by 9921622** [comment]
done

File: src/tags/tests/test_models.py [hunk 2]
```diff
@@ -179,6 +179,30 @@ def test_multiple_tags_per_quest_badges(self):
 
         self.assertEqual(expected_order, calculated_order)
 
+    def test_query_correct_quest(self):
```
Comment (tylerecouture) [inline src/tags/tests/test_models.py:182]: test naming convention.

File: src/tags/models.py [hunk 3]
```diff
@@ -24,7 +24,7 @@ def get_tags_from_user(user):
     BadgeAssertion = apps.get_model("badges", "BadgeAssertion")
 
     # tags related to user's quests
-    quest_ids = QuestSubmission.objects.filter(user=user).distinct().values_list('quest', flat=True)
+    quest_ids = QuestSubmission.objects.filter(user=user, is_approved=True).distinct().values_list('quest', flat=True)
```
Comment (tylerecouture) [inline src/tags/models.py:27]: `QuestSubmission.objects.filter(user=user, is_approved=True)`  Can you check if there is already a QuestSubmission queryset for this? Helps us reduce duplication

**Review by tylerecouture — CHANGES_REQUESTED** [review]

File: src/tags/models.py (hunk 3 above)
Comment (9921622) [reply src/tags/models.py:27]: done. used ```QuestSubmission.objects.all_approve(user=user).distinct()```

File: src/tags/tests/test_models.py (hunk 2 above)
Comment (9921622) [reply src/tags/tests/test_models.py:182]: calling the function to ```test_query_tag_from_approved_quest``` im not specifying the function being used as ```Tag_get_user_tags_and_xp_Tests``` says "Specialized TestClass for testing get_user_tags_and_xp function"

**Review by tylerecouture — APPROVED** [review]

**Pull request merged**